src/posts/aws-load-balancer-controller-with-existing-alb.md

```diff
@@ -0,0 +1,378 @@
+
+## AWS Load Balancer Controller
+
+The [AWS Load Balancer Controller](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/) is a Kubernetes controller that can manage the lifecycle of AWS Load Balancers, TargetGroups, Listeners (and Rules), and connect them with nodes (and pods) in your Kubernetes cluster. Check out the [how it works](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/how-it-works/) page for details on the design.
+
+![AWS-Load-Balancer-diagram-1](/blogs/aws-load-balancer-controller-with-existing-alb/aws-load-balancer-diagram-1.png)
+
+Looking back at our use case, we want to use an existing Application Load Balancer that is managed by Terraform. If you search for this online, you will most certainly find another lovely GitHub issue: [https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/228](https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/228)
+
+It’s a long issue, with lots of suggestions. Personally, I was not concerned with how much of the infrastructure we manage with Terraform vs Kubernetes; the primary goal was a simple solution that did not require too much customisation, and I found the idea of the [TargetGroupBinding](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/targetgroupbinding/targetgroupbinding/) Custom Resource Definition (CRD) quite appealing.
+
+### TargetGroupBinding Custom Resource Definition
+
+[TargetGroupBinding](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/targetgroupbinding/targetgroupbinding/) is a CRD that the AWS Load Balancer Controller installs. If you follow the most common use case and manage your ALBs with the AWS Load Balancer controller, it will create TargetGroupBindings under the hood even if you do not interact with them directly. Good news; it is a core feature of the AWS LB Controller, not an extension for people with an edge case. That gave me some confidence.
+
+It requires an existing TargetGroup and IAM policies to lookup and attach/detach targets to the TargetGroup. There’s a mention of the required IAM policies required [here](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/deploy/installation/#option-b-attach-iam-policies-to-nodes).
+
+Here’s a sample `TargetGroupBinding` manifest taken from the docs:
+
+```yaml
+apiVersion: elbv2.k8s.aws/v1beta1
+kind: TargetGroupBinding
+metadata:
+  name: ingress-nginx-binding
+spec:
+  # The service we want to connect to.
+  # We use the Ingress Nginx Controller so let's point to that service.
+  # NOTE: it was necessary for this TargetGroupBinding to be in the same namespace as the service.
+  serviceRef:
+    name: ingress-nginx
+    port: 80
+	# NOTE: need the ARN of the TargetGroup... Bit of a PITA.
```

Comment:
This looks like a tab, and is not aligned with the line below...

Reply:
<img width="940" alt="Screenshot 2022-10-17 at 16 10 13" src="https://user-images.githubusercontent.com/31657652/196185740-b437ac0d-44dd-4ca7-8b85-bb14b2aec819.png">
This is what it looks like on my local copy. It looks aligned, so I'm not sure how to fix it (same with the other tab issues).